substrate/frame/child-bounties/src/lib.rs

```diff
@@ -236,91 +260,106 @@ pub mod pallet {
 		#[pallet::call_index(0)]
-		#[pallet::weight(<T as Config>::WeightInfo::add_child_bounty(description.len() as u32))]
+		#[pallet::weight(<T as Config<I>>::WeightInfo::add_child_bounty(description.len() as u32))]
 		pub fn add_child_bounty(
 			origin: OriginFor<T>,
 			#[pallet::compact] parent_bounty_id: BountyIndex,
-			#[pallet::compact] value: BalanceOf<T>,
+			// TODO: `value` should pallet_treasury::AssetBalanceOf<T, I>,
+			// or better make pallet_treasury::AssetBalanceOf<T, I> and
+			// pallet_treasury::BalanceOf<T, I> same types for simplicity.
+			#[pallet::compact] value: BountyBalanceOf<T, I>,
 			description: Vec<u8>,
 		) -> DispatchResult {
 			let signer = ensure_signed(origin)?;
 
 			// Verify the arguments.
 			let bounded_description =
-				description.try_into().map_err(|_| BountiesError::<T>::ReasonTooBig)?;
-			ensure!(value >= T::ChildBountyValueMinimum::get(), BountiesError::<T>::InvalidValue);
+				description.try_into().map_err(|_| BountiesError::<T, I>::ReasonTooBig)?;
+			let asset_kind = Self::bounty_asset_kind(parent_bounty_id)?;
+			let native_amount =
+				<T as pallet_treasury::Config<I>>::BalanceConverter::from_asset_balance(
+					value,
+					asset_kind.clone(),
+				)
+				.map_err(|_| pallet_treasury::Error::<T, I>::FailedToConvertBalance)?;
+
+			ensure!(
+				native_amount >= T::ChildBountyValueMinimum::get(),
+				BountiesError::<T, I>::InvalidValue
+			);
 			ensure!(
-				ParentChildBounties::<T>::get(parent_bounty_id) <=
+				ParentChildBounties::<T, I>::get(parent_bounty_id) <=
 					T::MaxActiveChildBountyCount::get() as u32,
-				Error::<T>::TooManyChildBounties,
+				Error::<T, I>::TooManyChildBounties,
 			);
 
 			let (curator, _) = Self::ensure_bounty_active(parent_bounty_id)?;
-			ensure!(signer == curator, BountiesError::<T>::RequireCurator);
+			ensure!(signer == curator, BountiesError::<T, I>::RequireCurator);
 
 			// Read parent bounty account info.
 			let parent_bounty_account =
-				pallet_bounties::Pallet::<T>::bounty_account_id(parent_bounty_id);
-
+				pallet_bounties::Pallet::<T, I>::bounty_account_id(parent_bounty_id);
 			// Ensure parent bounty has enough balance after adding child-bounty.
-			let bounty_balance = T::Currency::free_balance(&parent_bounty_account);
-			let new_bounty_balance = bounty_balance
-				.checked_sub(&value)
-				.ok_or(Error::<T>::InsufficientBountyBalance)?;
-			T::Currency::ensure_can_withdraw(
-				&parent_bounty_account,
-				value,
-				WithdrawReasons::TRANSFER,
-				new_bounty_balance,
-			)?;
+			// Tiago: is this right? bounty_balance() returns T::Currency::free_balance of bounty
+			// native account id. but what I really want is to compare against the balance of the
+			// asset_kind of the parent bounty. as a plan B I sum the amounts of all child
+			// bounties and compare against the parent bounty amount.
+			let bounty_balance = pallet_bounties::Pallet::<T, I>::bounty_balance(parent_bounty_id);
```

Comment:
we cannot check it this way, the bounties balance might be on another chain. we can compare it against `bounty.amount` - `child_bounties.amount.sum`. or we keep track of remaining balance in `bounty.remaining_value`